execution/executor/src/chunk_executor.rs

```diff
@@ -264,189 +254,71 @@ impl<V: VMExecutor> ChunkExecutorInner<V> {
         Ok(chunk)
     }
 
-    // ************************* Chunk Executor Implementation *************************
-    fn enqueue_chunk_by_execution(
-        &self,
-        txn_list_with_proof: TransactionListWithProof,
-        verified_target_li: &LedgerInfoWithSignatures,
-        epoch_change_li: Option<&LedgerInfoWithSignatures>,
+    fn verify_chunk<Chunk: TransactionChunkWithProof + Sync>(
+        chunk: &Chunk,
+        ledger_info: &LedgerInfo,
+        first_version: Version,
     ) -> Result<()> {
-        let _timer = EXECUTE_CHUNK.start_timer();
-
-        let num_txns = txn_list_with_proof.transactions.len();
-        ensure!(num_txns != 0, "Empty transaction list!");
-        let first_version_in_request = txn_list_with_proof
-            .first_transaction_version
-            .ok_or_else(|| anyhow!("Non-empty chunk with first_version == None."))?;
-        let parent_state = self.commit_queue.lock().latest_state();
-        ensure!(
-            first_version_in_request == parent_state.next_version(),
-            "Unexpected chunk. version in request: {}, current_version: {:?}",
-            first_version_in_request,
-            parent_state.current_version,
-        );
```

Comment:
These are guaranteed if `list.verify()` passes with `Some(expected_version)` passed in, specifically, it's guranteed that 1. the list is not empty and 2. the list starts from `expected_version`